File: backend/app/models/glossary_entry.py
# app/models/glossary_entry.py

from typing import List
import uuid

class GlossaryEntry:
    def __init__(self,  
                 term: str, 
                 definition: str, 
                 term_audio_zip_urls: List[str] = None, 
                 definition_audio_zip_urls: List[str] = None,
                 user_id: int = None,
                 id: int = None):
        self.id = id if id else f"gt-{uuid.uuid4().hex[:12]}"
        self.user_id = user_id if user_id else ""
        self.term = term
        self.definition = definition
        self.term_audio_zip_urls = term_audio_zip_urls if term_audio_zip_urls else []
        self.definition_audio_zip_urls = definition_audio_zip_urls if definition_audio_zip_urls else []

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "user_id": self.user_id,
            "term": self.term,
            "definition": self.definition,
            "term_audio_zip_urls": self.term_audio_zip_urls,
            "definition_audio_zip_urls": self.definition_audio_zip_urls
        }

File: frontend/ReadBuddy/ReadBuddy/Models/Glossary/GlossaryEntry.cs
// ReadBuddy\Models\Glossary\GlossaryEntry.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;
using ReadBuddy.Models.TTS;

namespace ReadBuddy.Models.Glossary
{
    public class GlossaryEntry
    {
        [JsonPropertyName("id")]
        public string Id { get; private set; }

        [JsonPropertyName("user_id")]
        public string UserId { get; private set; }

        [JsonPropertyName("term")]
        public string Term { get; set; }

        [JsonPropertyName("definition")]
        public string Definition { get; set; }

        [JsonPropertyName("term_audio_zip_urls")]
        public List<string> TermAudioZipUrls { get; private set; }

        [JsonPropertyName("definition_audio_zip_urls")]
        public List<string> DefinitionAudioZipUrls { get; private set; }

        public TTSResult? TermTTS { get; set; } = null;
        public TTSResult? DefinitionTTS { get; set; } = null;

        public GlossaryEntry(string userId, string term, string definition,
                             List<string> termAudioZipUrls, List<string> definitionAudioZipUrls, string id)
        {
            UserId = userId;
            Term = term;
            Definition = definition;
            TermAudioZipUrls = termAudioZipUrls ?? new List<string>();
            DefinitionAudioZipUrls = definitionAudioZipUrls ?? new List<string>();
            Id = id;
        }
    }
}
